lib/file-source/src/file_server.rs

```diff
@@ -204,52 +203,53 @@ where
                 }
 
                 let mut bytes_read: usize = 0;
-                while let Ok(sz) = watcher.read_line(&mut line_buffer, self.max_line_bytes) {
-                    if sz > 0 {
-                        trace!(
-                            message = "read bytes.",
-                            path = field::debug(&watcher.path),
-                            bytes = field::debug(sz)
-                        );
-
-                        bytes_read += sz;
-
-                        if !line_buffer.is_empty() {
-                            lines.push((
-                                line_buffer.clone().into(),
-                                watcher.path.to_str().expect("not a valid path").to_owned(),
-                            ));
-                            line_buffer.clear();
-                        }
-                    } else {
-                        // Should the file be removed
-                        if let Some(grace_period) = self.remove_after {
-                            if watcher.last_read_success().elapsed() >= grace_period {
-                                // Try to remove
-                                match remove_file(&watcher.path) {
-                                    Ok(()) => {
-                                        self.emitter.emit_file_deleted(&watcher.path);
-                                        watcher.set_dead();
-                                    }
-                                    Err(error) => {
-                                        // We will try again after some time.
-                                        self.emitter.emit_file_delete_failed(&watcher.path, error);
-                                    }
-                                }
-                            }
-                        }
-
+                while let Ok(Some(line)) = watcher.read_line() {
+                    if line.is_empty() {
                         break;
                     }
+
+                    let sz = line.len();
+                    trace!(
```

Comment:
This could shift into an internal event perhaps?

Reply:
The way we do events here is a little different since it's its own crate, but not a bad idea. I'll make a note to do that in the next set of changes.